modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.server/src/main/scala/edu/gemini/seqexec/web/server/common/LogInitialization.scala

```diff
@@ -6,73 +6,77 @@ import java.util.logging.LogManager
 import scala.io.{Codec, Source}
 import scalaz._
 import Scalaz._
-import scalaz.effect.IO
+import scalaz.concurrent.Task
 
-trait LogInitialization {
-  sealed trait LogConfError
-  case class ConfigurationNotFound(s: String) extends LogConfError
-  case class LogDirNotFound(f: File) extends LogConfError
-  case object ConfigurationLoadingError extends LogConfError
+trait AppBaseDir {
+  /**
+    * Calculates the base dir of the application based on the location of "this" class jar file
+    * It will throw an exception if unable to find the base dir
+    */
+  def baseDir: Task[File] = Task.delay {
+    val clazz = this.getClass
+    val fileName = clazz.getResource(s"/${clazz.getName.replace(".", System.getProperty("file.separator"))}.class").getFile
 
-  type LogConf[A] = EitherT[IO, LogConfError, A]
```

Comment:
`Task` is just like `IO` but the `bind` effect allows the computation to be suspended and rescheduled on another thread, if the task has affinity for a thread pool (this is determined on construction). This is not directly observable, but it allows parallel execution of the applicative-like and traverse-like operations defined on `Nondeterminism`.

`.attempt` allows exceptions to be observed (and, as you note, ignored if you're not careful) at any point in the computation (not just at the top level), just as you can do with Haskell's `IO`. But it is possible to build up fine-grained combinators based on this primitive operation, as [here](https://github.com/tpolecat/doobie/blob/series/0.3.x/yax/core/src/main/scala/doobie/util/catchable.scala#L49-L68).